astropy_timeseries/binned.py

```diff
@@ -160,3 +160,88 @@ def __getitem__(self, item):
                                                  keys=self.groups._keys)
                 return out
         return super().__getitem__(item)
+
+    @classmethod
+    def read(self, filename, format, time_column=None, time_bin_start=None, time_bin_end=None,
+             time_bin_size_column=None, time_bin_size=None, time_bin_unit=None, *args, **kwargs):
+        """
+        Read and parse a file and returns a `astropy_timeseries.binned.BinnedTimeSeries`.
+
+        This function provides the access to the astropy unified I/O layer.
+        This allows easily reading a file in many supported data formats
+        using syntax such as::
+
+          >>> from astropy_timeseries.binned import BinnedTimeSeries
+          >>> dat = BinnedTimeSeries.read('table.dat', format='ascii', time_column='DATE')  # doctest: +SKIP
+
+        See also: http://docs.astropy.org/en/stable/io/unified.html
+
+        Parameters
+        ----------
+        filename: str
+            File to parse.
+        format : str
+            File format specifier.
+        time_column: str, optional
+            The name of the time column within the file.
+            This or ``time_bin_start`` must be passed in.
+        time_bin_start: str or list of str, optional
+            Can be a `astropy.time.Time` parseable string or a list of said strings.
+            This or ``time_column`` must be passed in.
+        time_bin_end: str, optional
+            The end time for the last binned data point.
+        time_bin_size: float or list, optional
+            The time step between each time index. If evenly sampled, you can specify
+            one single time step. If arbitrarily sampled, you can pass in a list.
+        time_bin_size_column: str, optional
+            If the time_bin_size is in the data file, use load it into the BinnedTimeSeries.
```

Comment:
I think to simplify that API we can assume that all the data is already in the file, so that you don't need to allow floats, quantities, Time and so on here but *just* column names. In which case, I think we'd want ``time_bin_start_column``, ``time_bin_end_column``, and ``time_bin_size_column``. We could leave ``time_bin_unit`` below since not all table formats support serializing units.

In addition, I think we want to add ``time_format`` and ``time_scale`` which should be passed to the ``Time`` class for the case where the start time and end time are read in.